fabtests/pytest/efa/test_efa_device_selection.py

```diff
@@ -0,0 +1,66 @@
+from subprocess import Popen, TimeoutExpired, PIPE
+import pytest
+
+# This test must be run in serial mode because it checks the hw counter
+@pytest.mark.serial
+@pytest.mark.functional
+def test_efa_device_selection(cmdline_args):
+    from efa.efa_common import efa_retrieve_hw_counter_value
+    from common import ClientServerTest
+
+    if cmdline_args.server_id == cmdline_args.client_id:
+        pytest.skip("EFA device selection test requires 2 nodes")
+        return
+
+    timeout = 60
+    process_timed_out = False
+
+    # This command returns a list of EFA domain names and its related info
+    command = "ssh {} fi_info -p efa".format(cmdline_args.server_id)
```

Comment:
It is better to extract this part into a separate function `get_efa_domain_names`.

since the function uses `ssh`,  better add retry on ssh connection failure.